http3/headers.go

```diff
@@ -178,25 +178,23 @@ func hostnameFromURL(url *url.URL) string {
 	return ""
 }
 
-func responseFromHeaders(headerFields []qpack.HeaderField) (*http.Response, error) {
+func responseFromHeaders(res *http.Response, headerFields []qpack.HeaderField) error {
```

Comment:
Why is this change necessary at all?

This function name doesn't seem accurate anymore. It would probably also help to add a comment.

Reply:
I rolled this change back. This was definitely evidence that I was going down the wrong path 😓